oak/server/rust/oak_runtime/src/lib.rs

```diff
@@ -1,645 +1,152 @@
-                );
-                return Err(OakStatus::ERR_INVALID_ARGS);
-            }
-        };
-        // Now find the corresponding entrypoint
-        let full_name = WasmEntrypointFullName {
-            config_name: config_name.to_string(),
-            entrypoint_name: entrypoint_name.to_string(),
-        };
-        let entrypoint: NodeMain = match config {
-            NodeConfiguration_oneof_config_type::wasm_config(_) => {
-                match self.entrypoint(&full_name) {
-                    Some(e) => e,
-                    None => {
-                        debug!(
-                            "{{{}}}: node_create('{}', '{}', {}) -> ERR_INVALID_ARGS",
-                            node_name, config_name, entrypoint_name, handle
-                        );
-                        return Err(OakStatus::ERR_INVALID_ARGS);
-                    }
-                }
-            }
-            NodeConfiguration_oneof_config_type::log_config(_) => {
-                log_node_main // note: entrypoint_name is ignored
-            }
-            NodeConfiguration_oneof_config_type::storage_config(_) => {
-                error!(
-                    "{{{}}}: Storage pseudo-Node not yet implemented!",
-                    node_name
-                );
-                return Err(OakStatus::ERR_INVALID_ARGS);
-            }
-        };
-
-        // Next, convert the creating Node's handle to a channel reference.
-        let half = match self.node_half_for_handle_dir(&node_name, handle, Direction::Read) {
-            Some(h) => h,
-            None => {
-                debug!(
-                    "{{{}}}: node_create('{}', '{}', {}) -> ERR_BAD_HANDLE",
-                    node_name, config_name, entrypoint_name, handle
-                );
-                return Err(OakStatus::ERR_BAD_HANDLE);
-            }
-        };
+        entrypoint: &str,
+        chan_reader: ChannelReader,
+    ) -> Result<(), OakStatus> {
+        let mut node_threads = runtime.node_threads.lock().unwrap();
 
-        // Create the new Node instance.
-        let new_node_name = self.next_node_name(&full_name);
-        let node = OakNode::new();
-        debug!(
-            "{{{}}}: add node running config '{}' entrypoint '{}'",
-            new_node_name, config_name, entrypoint_name
-        );
-        self.nodes.insert(new_node_name.to_string(), node);
+        if runtime.is_terminating() {
+            return Err(OakStatus::ERR_TERMINATED);
+        }
 
-        // Map the channel into the new Node's handle space.
-        let new_handle = self.node_add_half(&new_node_name, half);
+        let join_handle = runtime
+            .configurations
+            .get(config_name)
+            .ok_or(OakStatus::ERR_INVALID_ARGS)
+            .and_then(|conf| {
+                conf.new_instance(runtime.clone(), entrypoint.to_owned(), chan_reader)
+            })?;
+        node_threads.push(join_handle);
```

Comment:
The existing `oak_runtime` code has an internal but human-readable unique node name for each running node, and all the internal logging is set up to always use it in a common format. This is extremely helpful for debugging; can we replicate something similar here?

Reply:
I will re-enable this in this PR